service/biz/tpm_cert_verifier.go

```diff
@@ -111,39 +111,49 @@ func (tcv *DefaultTpmCertVerifier) VerifyIakAndIDevIDCerts(ctx context.Context,
 	}
 	log.InfoContext(ctx, "Successfully verified and parsed IAK cert")
 
-	iDevIDX509, err := VerifyAndParsePemCert(ctx, req.IDevIDCertPem, req.CertVerificationOpts)
-	if err != nil {
-		err = fmt.Errorf("failed to verify and parse IDevID cert: %v", err)
+	// Verify IAK cert subject serial and expected control card serial numbers match.
+	if diff := cmp.Diff(iakX509.Subject.SerialNumber, req.ControlCardID.GetControlCardSerial()); diff != "" {
```

Comment:
there's some fairly strong advice about NOT using cmp.Diff (or really cmp at all) in production code.
I don't know where this piece of code is supposed to be run, so can't help by suggesting that this is not a good place for this library to be used :(

Is this meant to be production code in a running service?
or is this more like a handy library/tool I can run to introspect how the process is going along?

Reply:
to be fair this looks like it wlaready existed in this code.. so..... ok, fine if you didn't add this I guess we should plan a cleanup later :)

Reply:
What is used instead of cmp.Diff? This will be in production code, but isn't yet since enrollz hasn't gone to prod. Will land this CL now since it's blocking enrollz work and file a bug to remove the cmp.Diff.  b/347135351